src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12129,92 +12166,35 @@ private bool ScanParenthesizedImplicitlyTypedLambda(Precedence precedence)
         {
             Debug.Assert(CurrentToken.Kind == SyntaxKind.OpenParenToken);
 
-            if (!(precedence <= Precedence.Lambda))
-            {
+            if (precedence > Precedence.Lambda)
                 return false;
-            }
-            //  case 1:  ( x ,
-            if (isParenVarCommaSyntax())
-            {
-                // Make sure it really looks like a lambda, not just a tuple
-                int curTk = 3;
-                while (true)
-                {
-                    var tk = this.PeekToken(curTk++);
 
-                    // skip  identifiers commas and predefined types in any combination for error recovery
-                    if (tk.Kind is not SyntaxKind.IdentifierToken and not SyntaxKind.CommaToken
-                        && !SyntaxFacts.IsPredefinedType(tk.Kind)
-                        && (this.IsInQuery || !IsTokenQueryContextualKeyword(tk)))
-                    {
-                        break;
-                    };
-                }
-
-                // ) =>
-                return this.PeekToken(curTk - 1).Kind == SyntaxKind.CloseParenToken &&
-                       this.PeekToken(curTk).Kind == SyntaxKind.EqualsGreaterThanToken;
-            }
-
-            //  case 2:  ( x ) =>
-            if (IsTrueIdentifier(this.PeekToken(1)))
-            {
-                // allow for       a) =>
-                var skipIndex = 2;
-
-                // Must have:     ) => 
-                if (this.PeekToken(skipIndex).Kind == SyntaxKind.CloseParenToken
-                    && this.PeekToken(skipIndex + 1).Kind == SyntaxKind.EqualsGreaterThanToken)
-                {
-                    return true;
-                }
-            }
+            //  ( x ) =>           or       ( ref x ) =>
+            //  ( x , ... ) =>     or       ( ref x , ...) =>
+            var index = 1;
 
-            //  case 3:  ( ) =>
-            if (this.PeekToken(1).Kind == SyntaxKind.CloseParenToken
-                && this.PeekToken(2).Kind == SyntaxKind.EqualsGreaterThanToken)
-            {
-                return true;
-            }
-
-            // case 4:  ( params
-            // This case is interesting in that it is not legal; this error could be caught at parse time but we would rather
-            // recover from the error and let the semantic analyzer deal with it.
-            if (this.PeekToken(1).Kind == SyntaxKind.ParamsKeyword)
-            {
-                return true;
-            }
-
-            return false;
-
-            bool isParenVarCommaSyntax()
+            while (true)
             {
-                var token1 = this.PeekToken(1);
+                var token = this.PeekToken(index++);
 
-                // Ensure next token is a variable
-                if (token1.Kind == SyntaxKind.IdentifierToken)
+                // Keep skipping modifiers, commas, and identifiers to consume the rest of the lambda arguments.
```

Comment:
>// Keep skipping modifiers, commas, and identifiers [](http://example.com/codeflow?start=16&length=51)

Does `ScanParenthesizedImplicitlyTypedLambda()` now return `true` for *explicitly-typed* lambda parameters when the types are simple identifiers? Did this method return `true` for those cases previously?

Reply:
commentd.